### ISSUE TITLE
Random Movies option doesn't show song banner at start of gameplay

### ISSUE THREAD

**Issue by shadowvfx** [stepmania/stepmania#1949]
Random Movies does not behave as it should. I'm referring to the Random Movies option under Options → Visual Options → Appearance Options → Random Gameplay Backgrounds

The behavior should act as follows: Song starts and the song's background image is displayed until the 1st notes hit the targets, then the background image should transition into RandomMovies. I have Stepmania 5.0.9 and can confirm that it was working in this build. However, in Stepmania 5.0.12, the behavior is broken. Instead of doing the above sequence, when the song starts, the background is immediately showing RandomMovies and does not show the song's background image.

**Comment by quietly-turning** [comment]
f1f1bd5ca69f5000a71404982a0bb693d5f9533b added Metrics that allowed SM5 themes to have more control over how Random Backgrounds behave during gameplay.  One of the new metrics was `RandomBGStartBeat` and the notes written at that time state `RandomBGStartBeat sets the beat of the music the first random bg occurs on.`

That commit opted to set `RandomBGStartBeat=-1000` in the *_fallback* theme's Metrics.ini in an effort to reasonably ensure that RandomMovies would start playing as quickly as possible in gameplay.

---

> The behavior should act as follows: Song starts and the song's background image is displayed until the 1st notes hit the targets, then the background image should transition into RandomMovies.

It sounds like you're describing ITG/OpenITG's random background behavior.  It's worth clarifying that ITG was a game and SM5 is more of an engine.  Many behaviors that were hardcoded in ITG have been made more flexible and left up to the discretion of themers in SM5.

I tested out the changes from f1f1bd5ca69f5000a71404982a0bb693d5f9533b in a theme just now and they seem to be functioning as intended.  You'll need to work with the author(s) of your theme(s).

I'll close this now in that it isn't a StepMania issue.

**Comment by shadowvfx** [comment]
So, if I understand this correctly, there is no way to set when RandomMovies starts based on the 1st note? I understand that you can set the beat using RandomBGStartBeat, but that will be universal for all songs, whether they have a long intro without notes or not. Am I correct?

**Comment by quietly-turning** [comment]
@shadowvfx – I revisited this issue and found that it is possible for themes to have the behavior you requested here.  I've added it to the Simply Love theme, and it will be included in the next formal release.

---

For other themers who find this thread years from now and are looking for code: https://github.com/quietly-turning/Simply-Love-SM5/commit/68426458887c7a72526a1afac4ac4cdbe60bb0ca

**Comment by coderjo** [comment]
I'm a bit curious what setting the RandomBGStartBeat metric to 0 will do. I also wonder if the default/fallback metrics should be changed back to what the behavior was before the metric was introduced.

**Comment by quietly-turning** [comment]
> I'm a bit curious what setting the RandomBGStartBeat metric to 0 will do.

Random backgrounds will start playing at beat 0.   This can vary from song to song depending on, I think, song offset.

The code I linked to will cause random backgrounds to start playing when the first note (or mine) crosses the targets.

_fallback's current setting `RandomBGStartBeat=-1000` seems to be an arbitrarily large negative value to ensure that random backgrounds are already playing when players see ScreenGameplay, but that's only my reading of it.  There's no context for why `-1000` was chosen that I've found.

---

> I also wonder if the default/fallback metrics should be changed back to what the behavior was before the metric was introduced.

I don't have strong feelings either way, personally.
